### PR TITLE
Fixed SSH support.

### PR DESCRIPTION
**Pull request by karya0**
This PR fixes the SSH functionality broken by #478.

### CODE REVIEW

**Comment by rohgarg** [comment]
@karya0: What is it that we broke with #478?

**Comment by karya0** [comment]
The argument parsing for dmtcp_sshd was not ported over correctly from 2.x. Check this:
https://github.com/dmtcp/dmtcp/pull/478/files#diff-efac9aa16eae3f242c03e30abf588260R172

Either it should have been `execvp(argv[6], &argv[6]);` or we should have used the `shift; shift;` trick.

**Comment by karya0** [comment]
It also indicates that we are not testing `ssh` in travis :(.

File: src/plugin/ipc/ssh/dmtcp_sshd.cpp [hunk 1]
```diff
@@ -105,31 +105,47 @@ dummySshdProcess(char *listenAddr)
...
+      shift;
+    } else {
+      break;
+    }
+  }
+
+  if (host == NULL) {
     printf("Missing --host argument");
-    assert(0);
+    exit(1);
```
Comment (gc00) [inline src/plugin/ipc/ssh/dmtcp_sshd.cpp]: `exit(1)` -> `exit(DMTCP_FAIL_RC)`

File: src/plugin/ipc/ssh/dmtcp_sshd.cpp [hunk 2]
```diff
@@ -105,31 +105,47 @@ dummySshdProcess(char *listenAddr)
...
   }
-  host = argv[2];
 
-  if (strcmp(argv[3], "--port") != 0) {
+  if (port == 0) {
     printf("Missing --port argument");
-    exit(0);
+    exit(1);
```
Comment (gc00) [inline src/plugin/ipc/ssh/dmtcp_sshd.cpp]: `exit(1)` -> `exit(DMTCP_FAIL_RC)`

**Comment by gc00** [comment]
I have two inline fixes of the form: exit(1) -> exit(DMTCP_FAIL_RC), Otherwise, LGTM.

I assume that the 2.5 branch is correct, and the fixes are only needed for 3.0. I'll put in a small PR for the 2.5 branch for adjusting the call to exit(1).

**Pull request merged**